build/Build.cs

```diff
@@ -0,0 +1,96 @@
+using Nuke.Common;
+using Nuke.Common.Execution;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tools.DotNet;
+using Nuke.Common.Utilities.Collections;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.IO.PathConstruction;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+using Nuke.Common.Tools.NuGet;
+using Nuke.OctoVersion;
+using OctoVersion.Core;
+
+[CheckBuildProjectConfigurations]
+class Build : NukeBuild
+{
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+    [Solution] readonly Solution Solution;
+    [NukeOctoVersion] readonly OctoVersionInfo OctoVersionInfo;
+
+    AbsolutePath LocalPackagesDirectory => RootDirectory / ".." / "LocalPackages";
+    AbsolutePath SourceDirectory => RootDirectory / "source";
+    AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
+    AbsolutePath PublishDirectory => RootDirectory / "publish";
+    AbsolutePath AssetDirectory => RootDirectory / "BuildAssets";
+
+    Target Clean =>
+        _ => _
+            .Before(Restore)
+            .Executes(() =>
+            {
+                EnsureCleanDirectory(ArtifactsDirectory);
+                EnsureCleanDirectory(PublishDirectory);
+                SourceDirectory.GlobDirectories("**/bin", "**/obj").ForEach(EnsureCleanDirectory);
+            });
+
+    Target Restore =>
+        _ => _
+            .DependsOn(Clean)
+            .Executes(() =>
+            {
+                DotNetRestore(settings => settings.SetProjectFile(Solution).SetVersion(OctoVersionInfo.NuGetVersion));
```

Comment:
nitpick: I prefer to see these written like you've done with the DotNetPack settings:

```suggestion
                DotNetRestore(settings => settings
                    .SetProjectFile(Solution)
                    .SetVersion(OctoVersionInfo.NuGetVersion));
```